src/components/VideoPlayer/IconButton.tsx

```diff
@@ -1,42 +1,23 @@
-import PropTypes from 'prop-types';
 import React from 'react';
+import type {GestureResponderEvent, StyleProp, ViewStyle} from 'react-native';
 import Icon from '@components/Icon';
 import PressableWithFeedback from '@components/Pressable/PressableWithFeedback';
 import Tooltip from '@components/Tooltip';
 import useThemeStyles from '@hooks/useThemeStyles';
-import stylePropTypes from '@styles/stylePropTypes';
-
-const propTypes = {
-    // use IconAsset as soon as it will be migrated to TS
-    // eslint-disable-next-line react/forbid-prop-types
-    src: PropTypes.any.isRequired,
-
-    onPress: PropTypes.func,
-
-    fill: PropTypes.string,
-
-    tooltipText: PropTypes.string,
-
-    style: stylePropTypes,
-
-    hoverStyle: stylePropTypes,
-
-    small: PropTypes.bool,
-
-    shouldForceRenderingTooltipBelow: PropTypes.bool,
-};
-
-const defaultProps = {
-    fill: 'white',
-    onPress: () => {},
-    style: {},
-    hoverStyle: {},
-    small: false,
-    tooltipText: '',
-    shouldForceRenderingTooltipBelow: false,
+import type IconAsset from '@src/types/utils/IconAsset';
+
+type IconButtonProps = {
+    src: IconAsset;
+    onPress?: (event?: GestureResponderEvent | KeyboardEvent) => void | Promise<void>;
+    fill?: string;
+    tooltipText?: string;
+    style?: StyleProp<ViewStyle>;
+    hoverStyle?: StyleProp<ViewStyle>;
+    small?: boolean;
+    shouldForceRenderingTooltipBelow?: boolean;
 };
 
-function IconButton({src, fill, onPress, style, hoverStyle, tooltipText, small, shouldForceRenderingTooltipBelow}) {
+function IconButton({src, fill = 'white', onPress = () => {}, style = {}, hoverStyle = {}, tooltipText = '', small = false, shouldForceRenderingTooltipBelow = false}: IconButtonProps) {
```

Comment:
Let's remove empty objects for styles, and maybe onPress too?

(styles work for falsy values too, and onPress can be undefined and we can add optional chaining like: `onPress?.()`)